schemas/json/webhook.json

```diff
@@ -0,0 +1,77 @@
+{
+	"title": "Web hook",
+	"description": "This web hook allows us to send a JSON object from the service portal",
+	"type": "object",
+	"required": [
+		"url"
+	],
+	"definitions": {
+            "Authentications": {
+                "title": "Authentications",
+                "type": "string",
+                "anyOf": [
+                     {
+                       "type": "string",
+                       "enum": [
+                          "oauth"
+                       ],
+                       "title": "OAuth 2.0"
+                     },
+                     {
+                        "type": "string",
+                        "enum": [
+                            "basic"
+                        ],
+                        "title": "Basic Authentication"
+                     },
+                     {
+                        "type": "string",
+                        "enum": [
+                            "none"
+                        ],
+                        "title": "No Authentication needed"
+                     }
+                ]
+            }
+         },
+	"properties": {
+		"url": {
+			"type": "string",
+			"title": "URL",
+			"description": "The URL which will be receving this request",
+			"pattern": "^(http|https)://*"
+		},
+		"verify_ssl": {
+			"type": "boolean",
+			"default": true,
+			"title": "Verify Server Certificate"
+		},
+		"secret": {
+			"type": "string",
+			"title": "Secret",
+			"description": "If specified we will create a HMAC signature of the body with the secret and include it in the HTTP Header X-Service-Portal-Signature"
+		},
+		"authentication": {
```

Comment:
Should we have a definition for the type of hook they are registering for? e.g. Approval notifications or Provisioning payloads?

Reply:
@syncrou We would know the context inside of the State Machine that does the provisioning it knows which web hook to use based on the context.

If the Portfolio Item say has 2 JSONB columns
 - Pre_Provision_Webhook
 - Post_Provision_Webhook

And it has the above set of parameters defined in the Payload

When we are in the Pre Provision Step of the State Machine we would create a payload that could look like

``` ruby
{
   'name' => 'Maria DB',
   'state' => 'Pre Provision',
  'requester' => 'Fred Flintstone',
  'timestamp' => '2018-10-12T14:12:56Z'
}
```

These parameters are not defined in the WebHook the body is constituted at runtime by us
The WebHook parameters used would be from Pre_Provision_Webhook

After we have completed the Provision we would create another body
``` ruby
{
   'name' => 'Maria DB',
   'state' => 'Post Provision',
  'requester' => 'Fred Flintstone',
  'timestamp' => '2018-10-12T14:12:56Z'
 'result' => 'Failure',
'errorMessage' => 'Not enough disk space'
}
```
These parameters are not defined in the WebHook the body is constituted at runtime by us
The WebHook parameters used would be from Post_Provision_Webhook

So technically the data could be sent to 2 different systems for pre and post steps